aten/src/ATen/native/mps/operations/LinearAlgebra.mm

```diff
@@ -560,6 +561,90 @@ bool use_metal_mm(const Tensor& self, const Tensor& other, const Tensor& output)
   return out;
 }
 
+static void lu_factor_stub_mps(const Tensor& LU, const Tensor& pivots, const Tensor& info, bool compute_pivots) {
+  TORCH_CHECK(compute_pivots, "linalg.lu_factor: LU without pivoting is not implemented on MPS device");
+
+  auto LU_ = LU.is_contiguous() ? LU : LU.contiguous();
+  auto pivots_ =
+      at::zeros_like(pivots, pivots.options().dtype(ScalarType::Int).memory_format(MemoryFormat::Contiguous));
+
+  id<MTLBuffer> luBuffer = getMTLBufferStorage(LU_);
+  id<MTLBuffer> pivotsBuffer = getMTLBufferStorage(pivots_);
+  id<MTLDevice> device = MPSDevice::getInstance()->device();
+  MPSStream* mpsStream = getCurrentMPSStream();
+
+  uint64_t luRows = LU.size(-2);
+  uint64_t luCols = LU.size(-1);
+  uint64_t pivotCols = pivots.size(-1);
+  auto batchSize = batchCount(LU);
+  uint64_t luElemSize = LU.element_size();
+
+  dispatch_sync(mpsStream->queue(), ^() {
+    @autoreleasepool {
+      mpsStream->endKernelCoalescing();
```

Comment:
You don't need to end coalescing. Were you running into any correctness issues without this ?

Reply:
Removed it, thanks!